hibernate-reactive-routes-quickstart/src/main/java/org/acme/hibernate/reactive/FruitsRoutes.java

```diff
@@ -1,123 +1,112 @@
 package org.acme.hibernate.reactive;
 
-import javax.enterprise.context.ApplicationScoped;
+import static io.quarkus.vertx.web.Route.HandlerType.FAILURE;
+import static io.vertx.core.http.HttpMethod.DELETE;
+import static io.vertx.core.http.HttpMethod.GET;
+import static io.vertx.core.http.HttpMethod.POST;
+import static io.vertx.core.http.HttpMethod.PUT;
+
+import java.util.NoSuchElementException;
+
 import javax.inject.Inject;
 
 import org.hibernate.reactive.mutiny.Mutiny;
+import org.jboss.logging.Logger;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.databind.ObjectMapper;
+import io.quarkus.vertx.web.Body;
+import io.quarkus.vertx.web.Param;
+import io.quarkus.vertx.web.ReactiveRoutes;
 import io.quarkus.vertx.web.Route;
 import io.quarkus.vertx.web.RouteBase;
 import io.smallrye.mutiny.Multi;
 import io.smallrye.mutiny.Uni;
 import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
 import io.vertx.ext.web.RoutingContext;
 
-import static io.vertx.core.http.HttpMethod.DELETE;
-import static io.vertx.core.http.HttpMethod.GET;
-import static io.vertx.core.http.HttpMethod.POST;
-import static io.vertx.core.http.HttpMethod.PUT;
-
 /**
  * An example using Vert.x Web, Hibernate Reactive and Mutiny.
  */
-@ApplicationScoped
 @RouteBase(path = "/fruits", produces = "application/json")
 public class FruitsRoutes {
 
-    @Inject
-    ObjectMapper mapper;
+    private static final Logger LOGGER = Logger.getLogger(FruitsRoutes.class.getName());
 
     @Inject
     Mutiny.Session session;
 
-    /**
-     * Convert an object into a JSON string.
-     */
-    private String toJSON(Object object) {
-        try {
-            return mapper.writeValueAsString(object);
-        } catch (JsonProcessingException e) {
-            throw new IllegalArgumentException(e);
-        }
-    }
-
     @Route(methods = GET, path = "/")
-    public Multi<Fruit> getAll(RoutingContext rc) throws Exception {
-        return session.createNamedQuery(Fruit.FIND_ALL, Fruit.class).getResults();
+    public Multi<Fruit> getAll() throws Exception {
+        return ReactiveRoutes.asJsonArray(session.createNamedQuery(Fruit.FIND_ALL, Fruit.class).getResults());
```

Comment:
Is `ReactiveRoutes.asJsonArray(` required now?

Reply:
I think that it was required even before but the test does not validate the proper syntax of json array...

Reply:
Yes, it required (For now) to creates an asynchronous json array. Each item is written in the response when ready (and sent to the client). 

We are improving this by inferring this from the `produces` attribute. But it's WIP...

Reply:
It surprised me because when I write
```
    @Route(methods = GET, path = "/")
    public Uni<List<Fruit>> getAll(RoutingContext rc) throws Exception {
        return session.createNamedQuery( Fruit.FIND_ALL, Fruit.class).getResultList();
    }
```

It works as expected. I'm wondering if it wouldn't look better to return a `Uni<List>`?

Reply:
yes it looks better, but the quickstart is supposed to teach to people "the right way".

@cescoffier are you optimistic about your WIP to be doable? If so, maybe it's ok for the example to already show the simpler syntax?  We're not working to win a benchmark war _yet_ ;)

Reply:
The right way is the one that works ;)

Reply:
Multi<X> and Uni<List<X>> don't have the same semantic. Multi<X> writes the X into the response when they are received, avoiding the items to be stored in memory. Uni<List<X>>> builds the list and write it in one "batch". We recommend List<X> is it fits in memory and if the items of the list can be retrieved in one _batch_. In which case are we with Hibernate Reactive?

Reply:
Hibernate Reactive at the moment only works with `Uni<List>` and the Multi returned by `query.getResults()` is just the Uni converted to a Multi.

That said, this quickstart is a simple case that returns a bunch of fruits, picking a Multi or a Uni in the API is more a question of what features we want to show.

The reason I used `Multi<Fruit>` is because it looks better and that was my goal for the quickstart.
But if it needs extra handling, I would return a Uni and add a comment to explain why it makes sense in this case to use it (instead of a Multi) and maybe create another example that shows when it makes sense to use a Multi.

Reply:
`Uni<List<Fruit>>` works fine and makes probably more sense for this use case (unless there's a plan to support `Multi` "properly").

@cescoffier @Sanne Should we change the return type to `Uni<List<Fruit>>`? And if so what kind of comment should we add? _"Hibernate Reactive does not support this yet"_ does not sound good ;-).

Reply:
From a reactive standpoint, returning a Multi is better.

Reply:
As I said, there are also good arguments to return a `Uni<List>` so I don't mind.

Reply:
thanks @cescoffier , let's use what looks best then.

> Should we change the return type to `Uni<List<Fruit>>`? And if so what kind of comment should we add? _"Hibernate Reactive does not support this yet"_ does not sound good ;-).

I don't feel a comment is necessary. What is triggered when using this API is a single SQL statement which expects a reasonably sized result set. For example, here we'll load all of 4 fruits from the DB.. there's no need to use a stream for this, in fact it would be less efficient as it's more chatty to the DB, while the whole resultset is likely to fit in a single network packet.

For streaming results, Hibernate offers different APIs which people are encouraged to use when the expected size of the results is significant. (while getResultsList() is intentionally returning a `Uni<List<R>>` ).

Reply:
P.S. I see that the alternative API which I referred to - one which produces a "real Multi" - isn't around yet, but I expect we'll be able to make one based on the scrollable API of ORM - wich is backed by server side cursors.

Reply:
@gavinking ^

Reply:
> let's use what looks best then.

`Uni<List<Fruit>>` then? ;-)

Reply:
> "Hibernate Reactive does not support this yet" does not sound good 

I don't think this is a Hibernate Reactive issue, it's a general issue when designing the API. Not saying that we need to add a comment but something similar to this would tell the user the differences:
```
In this case it makes sense to return a Uni<List<Fruit>> because we return a reasonable amount of results. Consider return a Multi<Fruit> for streaming results (link to example).
```

Reply:
For everybody interested, this is the issue explaining why Hibernate Reactive hasn't a proper Multi return method: https://github.com/hibernate/hibernate-reactive/issues/153

Having some feedback is always nice.